### PR TITLE
Settings: adjust Publish confirmation for Gutenberg enabled sites

### LINKED ISSUE #29627: Gutenberg: Respect site writing settings
Site settings affecting the editor aren't currently used by Gutenberg, as Gutenberg uses a different store (and "ruleset") than Calypso.

For example, Calypso has a "show publish confirmation" option in the Writing settings:
screenshot 2018-12-19 at 17 12 19

Gutenberg has such confirmation by default:
screenshot 2018-12-19 at 17 12 05

This option is handled entirely by Gutenberg, and can be toggled with the "Enable Pre-publish Checks" check in the Options dialog accessible from the More menu.

As I see it, we can solve this in two ways:
- Update Gutenberg so that it syncs the "pre-publish checks" option with Calypso's "show publish confirmation".
- For opted-in to Gutenberg users, remove the option from Calypso's writing settings, and rely only on the Gutenberg's own.

#### Note
We should run an actual audit of Calypso settings and see if there are other discrepancies with Gutenberg.

### PR DESCRIPTION
**Pull request by retrofox**
It replaces the "Show publish confirmation" toggle option from the `Settings/Writing` with an informative text whether the current editor of the site is the Block Editor, delegating the control of this feature to the editor.

Fixes #29627 

### Classic Editor

If the Classic editor is the current one, the Settings/Writing page should show:

and when the post is going to be published it should show the following advice:

### Block Editor

If the Block Editor is activated, the Settings/Writing page should show just a message:

image

_(Adding `Needs Design Review` and `Needs Copy Review` )

So now when it's going to be published the Block editor will take over of warning to customers:

### CODE REVIEW

File: client/my-sites/site-settings/composing/publish-confirmation.jsx [hunk 1]
```diff
@@ -87,6 +106,7 @@ export default connect(
 			siteId,
 			fetchingPreferences: isFetchingPreferences( state ),
 			publishConfirmationEnabled: isConfirmationSidebarEnabled( state, siteId ),
+			showPublishFlow: getSelectedEditor( state, siteId ) === 'gutenberg',
```
Comment (obenland) [inline client/my-sites/site-settings/composing/publish-confirmation.jsx]: In other instances where `getSelectedEditor()` is being used, I see it used together with `isGutenbergEnabled()` or `isCalypsoifyGutenbergEnabled()`. @Copons How should it be used ideally? Would these other checks be necessary here?

File: client/my-sites/site-settings/composing/publish-confirmation.jsx [hunk 2]
```diff
@@ -43,7 +45,24 @@ class PublishConfirmation extends Component {
...
+
+		if ( showPublishFlow ) {
+			return (
+				<FormFieldset>
+					<FormLabel>{ translate( 'Show publish confirmation' ) }</FormLabel>
+					<FormSettingExplanation isIndented>
+						{ translate(
+							'The confirmation step for double-checking your content before publishing ' +
+								'now is handled by the new block editor.' +
+								'To show the publish confirmation with the block editor, ' +
```
Comment (obenland) [inline client/my-sites/site-settings/composing/publish-confirmation.jsx]: Do we need a line break or a whitespace before `To`?

File: client/my-sites/site-settings/composing/publish-confirmation.jsx [hunk 3]
```diff
@@ -43,7 +45,24 @@ class PublishConfirmation extends Component {
...
+								'click the editor menu, then "Options", ' +
+								'and check "Enable Pre-publish Checks.'
```
Comment (obenland) [inline client/my-sites/site-settings/composing/publish-confirmation.jsx]: Looks like it's missing a closing `"` here?

File: client/my-sites/site-settings/composing/publish-confirmation.jsx (hunk 1 above)
Comment (Copons) [reply client/my-sites/site-settings/composing/publish-confirmation.jsx]: Good question!
I'm trying to remember all the nuances of those selectors, but it's really hard... 😅 

There might be edge cases where users are using Guten in wp-admin on site types that Calypso doesn't support Guten for (e.g. Jetpack and VIP?).
This means that someone might be using Guten in wp-admin and Classic in Calypso. They should still be able to control the publish flow for Classic Calypso.

I'd say let's keep the additional check in place just to be sure and avoid possible UX inconsistencies across site types and user preferences, as combinations are too many to be reliably tested:
`showPublishFlow: isGutenbergEnabled( state, siteId ) && getSelectedEditor( state, siteId ) === 'gutenberg'`

File: client/my-sites/site-settings/composing/publish-confirmation.jsx [hunk 4]
```diff
@@ -43,7 +46,24 @@ class PublishConfirmation extends Component {
...
+
+		if ( showPublishFlow ) {
+			return (
+				<FormFieldset>
+					<FormLabel>{ translate( 'Show publish confirmation' ) }</FormLabel>
+					<FormSettingExplanation isIndented>
+						{ translate(
+							'The confirmation step for double-checking your content before publishing ' +
+								'now is handled by the new block editor.' +
+								'To show the publish confirmation with the block editor, ' +
```
Comment (obenland) [inline client/my-sites/site-settings/composing/publish-confirmation.jsx]: @Automattic/editorial When you get a chance, would you mind looking over this description?

Comment (retrofox) [reply client/my-sites/site-settings/composing/publish-confirmation.jsx]: Thanks, Konstantin

Comment (kwight) [reply client/my-sites/site-settings/composing/publish-confirmation.jsx]: @Automattic/editorial or maybe @michelleweber , could we ask you for input on the content here?

**Comment by kwight** [comment]
Works and looks great to me – we just need to update content according to Editorial 🚢

**Comment by gwwar** [comment]
@vindl can someone take this one over while @retrofox is out?

**Comment by simison** [comment]
Does the text show up also for users who've never had classic editor? It can be little confusing for them :-)

File: client/my-sites/site-settings/composing/publish-confirmation.jsx [hunk 5]
```diff
@@ -43,7 +46,24 @@ class PublishConfirmation extends Component {
...
+		const { fetchingPreferences, translate, showPublishFlow } = this.props;
+
+		if ( showPublishFlow ) {
+			return (
+				<FormFieldset>
+					<FormLabel>{ translate( 'Show publish confirmation' ) }</FormLabel>
+					<FormSettingExplanation isIndented>
+						{ translate(
+							'The confirmation step for double-checking your content before publishing ' +
+								'now is handled by the new block editor.' +
```
Comment (simison) [inline client/my-sites/site-settings/composing/publish-confirmation.jsx]: Note missing space after `.`:

> ...block editor.To show the publish

**Comment by vindl** [comment]
> @vindl can someone take this one over while @retrofox is out?

Sure, I'll take it.

**Comment by vindl** [comment]
I tried to simplify the text a bit, but I'd still prefer a review from @Automattic/editorial if possible.

**Comment by kristastevens** [comment]
> I tried to simplify the text a bit, but I'd still prefer a review....

Hi @vindl, happy to have a look. Is there a revised version of this screen with your simplified text? Can you share a screenshot?

**Comment by vindl** [comment]
Hi @kristastevens, here is the screenshot of the new text:

Screenshot 2019-04-08 at 10 46 35

**Comment by kristastevens** [comment]
Thanks @vindl  -- how about: 

The Block Editor handles the Publish confirmation setting. To enable it, go to Options under the Ellipses menu in the Editor and check "Enable Pre-publish checks."

**Comment by vindl** [comment]
>Thanks @vindl -- how about:
> The Block Editor handles the Publish confirmation setting. To enable it, go to Options under the > Ellipses menu in the Editor and check "Enable Pre-publish checks."

Sounds great! Updated in e03bc8f985e9feb9f4eb29574aeb969c0f1b691f.

**Pull request merged**